src/main/java/org/saltyrtc/client/signaling/Signaling.java

```diff
@@ -502,9 +500,11 @@ public void onDisconnected(WebSocket websocket,
                     }
                 }
                 // Note: Don't check for signaling state here, it will already have been resetted.
-                if (closeCode != CloseCode.HANDOVER) {
-                    Signaling.this.salty.events.close.notifyHandlers(new CloseEvent(closeCode));
-                    setState(SignalingState.CLOSED);
+                synchronized (this) {
```

Comment:
Let's just make the whole method synchronised.

Reply:
Done.